proposals/push-constants.md

```diff
@@ -0,0 +1,74 @@
+# ImmediateData
+
+Status: **Draft**
+
+Last modified: 2024-05-30
+
+Issue: #75
+
+# Requirements
+
+No special requirements.
+
+# WGSL
+
+
+## Address Spaces
+
+| Address space | Sharing among invocations | Default access mode | Notes |
+| --- | --- | --- | --- |
+| `immediate_data` | Invocations in [shader stage](https://www.w3.org/TR/WGSL/#shader-stages) | [read](https://www.w3.org/TR/WGSL/#access-read) | For [uniform buffer](https://www.w3.org/TR/WGSL/#uniform-buffer) variables exclude [array types](https://www.w3.org/TR/WGSL/#array-types) variable or [structure types](https://www.w3.org/TR/WGSL/#struct-types) variable contains [array types](https://www.w3.org/TR/WGSL/#array-types) attributes |
+
+
+## Variable and Value Declarations
```

Comment:
Would be nice to have sample code for what an immediate_data variable declaration would look like. (IIUC there is no `@location` or anything like that - it would make that clear.)